src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -347,7 +370,14 @@ protected function instantiateObject(array &$data, string $class, array &$contex
             $params = [];
             foreach ($constructorParameters as $constructorParameter) {
                 $paramName = $constructorParameter->name;
-                $key = $this->nameConverter ? $this->nameConverter->normalize($paramName, $class, $format, $context) : $paramName;
+                $key = $paramName;
+                if (\count($this->nameConverters) > 0) {
+                    $nameConverter = $this->nameConverters[$context[static::NAME_CONVERTER] ?? null] ?? reset($this->nameConverters);
```

Comment:
If the given name converter in context doesn't correspond to one of the name converters, the first name converter is used instead.
The user is not warned, but I think it's the right approach. WDYT?

Reply:
an exception should be thrown instead to me
having a service locator would naturally throw a service-bot-found exception btw.